File: homework/HW10/JavaVersion/LevelOrderTraversal.java
package homework.HW10.JavaVersion;
import java.util.*;


public class LevelOrderTraversal{

    //levelOrder Method
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> order = new ArrayList<List<Integer>>();

        if(root == null){
            return order;
        }

        Queue<TreeNode> queue = new LinkedList<>();
        
        queue.add(root);

        //Loop to insert nodes into a list, then insert List into Final List
        while(!queue.isEmpty()){            //Loop until every node is in a list
            int levelSize = queue.size();
            List<Integer> levelNodes = new ArrayList<>();

            //Inserting # of nodes based on length of queue
            for(int i = 0; i < levelSize; i++){
                TreeNode curr = queue.poll();
                levelNodes.add(curr.val);
                if(curr.left != null){      //Insert curr node's children(s) into queue
                    queue.add(curr.left);
                }
                if(curr.right != null){
                    queue.add(curr.right);
                } 
            }
            order.add(levelNodes);       //Insert levelNodes into final list
        }

        return order;
    }

    public static void main(String[] args) {
        
        TreeNode x = null;
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(3);
        root.left.left = new TreeNode(1);
        root.right = new TreeNode(8);
        root.right.left = new TreeNode(5);
        root.right.right = new TreeNode(9);

        LevelOrderTraversal test = new LevelOrderTraversal();

        //Test Case (root is null)
        List<List<Integer>> ans = test.levelOrder(x);
        System.out.println("Level Order (null): " + ans);

        //Test Case
        ans = test.levelOrder(root);
        System.out.println("Level Order: " + ans);

    }

}

File: homework/HW10/PythonVersion/LevelOrderTraversal.py
from typing import List, Optional
from TreeNode import TreeNode 

tree = TreeNode     #Variable to use TreeNode defs

#levelOrder definition
def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
    order = []

    if root is None:
        return order
    
    queue = []      

    queue.append(root)  #Insert root node into queue
    
    #Loop to insert nodes into a List, then insert List to Final List
    while len(queue) > 0:
        levelSize = len(queue)
        levelNodes = []

        #Inserting # of nodes based on length of queue
        for i in range(levelSize):
            curr = queue.pop(0)         #Making list like a queue by popping index 0
            levelNodes.append(curr.val)
            if curr.left is not None:
                queue.append(curr.left)
            if curr.right is not None:
                queue.append(curr.right)

        order.append(levelNodes)     #Insert nodes list into final list 
    
    return order

#Construct nodes
x = tree.TreeNode(None)
root = tree.TreeNode(4)
root.left = tree.TreeNode(3)
root.left.left = tree.TreeNode(1)
root.right = tree.TreeNode(8)
root.right.left = tree.TreeNode(5)
root.right.right = tree.TreeNode(9)

#Test Case (root is None)
test = levelOrder(x,x)
print("Level Order (null node): ", end = "")
print(test)

#Test Case 
test = levelOrder(root,root)
print("Level Order: ", end="")
print(test)
